Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp

```diff
@@ -3015,21 +3023,22 @@ namespace AzToolsFramework
 
             m_actionManagerInterface->RegisterCheckableAction(
                 EditorMainWindowActionContextIdentifier,
-                "o3de.action.edit.transform.scale",
+                actionIdentifier,
                 actionProperties,
                 [&]()
                 {
                     SetTransformMode(Mode::Rotation);
                 },
                 [&]() -> bool
                 {
-                    return GetTransformMode() == Mode::Rotation;
+                    return GetTransformMode() == Mode::Scale;
                 }
```

Comment:
Good spot! 😝 Also while we're here could we change the capture from `[&]` to `[this]` to make it more specific to what's captures (same in the other cases too)